New mini book:  OS setup for Fortran developement (#152)
fortran-lang/fortran-lang.org: learn/os_setup/ides.md

---
layout: book
title: IDEs
permalink: /learn/os_setup/ides
---

An IDE (Integrated Development Environment) refers to a complete software development environment, where, all those packages that come as external plug-ins in Text-Editors, are already integrated within the software. An IDE is usually optimized towards a specific set of languages. For example it is very common for IDEs to advertise themselves towards either compiled or interpreted languages, or even towards a single language, or depending on the application developed, like scientific or web development. 

> IDEs are very recommended to beginner programmers, since it is possible to start coding with minimum effort after installation. However, it is quite common, in professional environments, individual developers and teams alike, to choose an IDE for large projects because of some unique features that they may offer over a Text-Editor.

Reviewer: ```suggestion
IDEs are recommended for beginner programmers, since it is possible to start coding with minimum effort after installation. However, it is quite common, in professional environments, individual developers and teams alike, to choose an IDE for large projects because of some unique features that they may offer over a Text-Editor.
```

A list of popular IDEs that provide good Fortran support is provided in alphabetical order:

> - [AbsoftTools](https://www.absoft.com/technology/absofttools-fortran-ide/) is commersial product, cross-platform, comes with Absoft's own Fortran compiler.

Reviewer: ```suggestion
- [AbsoftTools](https://www.absoft.com/technology/absofttools-fortran-ide/): commercial product, cross-platform, comes with Absoft's own Fortran compiler.
```

> - [Code::Blocks](http://www.codeblocks.org/) is free product, cross-platform and supports multiple compilers. Plenty of tutorials online on how to install it and add a Fortran Compiler. As well as a recent presentation, [here](https://www.youtube.com/watch?v=M1RwVGGSAgE&ab_channel=FortranCon), of its capabilities and future prospects at the International Fortran Conference 2020 by its main developer. 

Reviewer: ```suggestion
- [Code::Blocks](http://www.codeblocks.org/): free product, cross-platform and supports multiple compilers. Plenty of tutorials online on how to install it and add a Fortran Compiler. As well as a recent presentation, [here](https://www.youtube.com/watch?v=M1RwVGGSAgE&ab_channel=FortranCon), of its capabilities and future prospects at the International Fortran Conference 2020 by its main developer. 
```

> - [Eclipse-Photran](https://marketplace.eclipse.org/content/photran-fortran-ide-eclipse) *Photran is an IDE and refactoring tool for Fortran based on Eclipse and the C/C++ Development Tools*. Is free product, cross-platform and supports multiple compilers.

Reviewer: ```suggestion
- [Eclipse-Photran](https://marketplace.eclipse.org/content/photran-fortran-ide-eclipse) *Photran is an IDE and refactoring tool for Fortran based on Eclipse and the C/C++ Development Tools*. A free product, cross-platform and supports multiple compilers.
```

> - [Geany](https://www.geany.org/), is free product, cross platform and supports multiple compilers.

Reviewer: ```suggestion
- [Geany](https://www.geany.org/): a free product, cross platform and supports multiple compilers.
```

- [NAG Fortran Builder](https://www.nag.com/content/nag-fortran-builder-0) is a commercial product, available in Windows and MacOS and supports NAG's own Fortran Compiler.

> - [Plato](https://www.silverfrost.com/16/ftn95/plato.aspx) is commersial product, cross-platform, comes with Silverfrost's Fortran FTN95 own Fortran compiler.

Reviewer: ```suggestion
- [Plato](https://www.silverfrost.com/16/ftn95/plato.aspx) is a commercial product, cross-platform, comes with Silverfrost's Fortran FTN95 own Fortran compiler.
```

- [SimplyFortran](https://simplyfortran.com/) is a commercial product, cross-platform with support of the GNU Fortran compiler.

> - [Visual Studio](https://visualstudio.microsoft.com/) its *Community Edition 2019* is a free product, available only in Windows and supports only the commercial* Intel's Fortran compiler. 

Reviewer 2: The first few sentences need some editing: 

- "environment, where, all ..., here are already" should be: "environment, where all ..., are already"
- "For example [it] is very common [for] IDEs ..." (Perhaps even a more rigorous editing ;))
- "[IDEs] are [v]ery often recommended ..."
- "due to some ..." -> "because of some ..."

> - [Visual Studio](https://visualstudio.microsoft.com/) its *Community Edition 2019* is a free product, available only in Windows and supports only the commercial* Intel's Fortran compiler. 

Author: @certik I already have vscode mentioned at the text-editors page.
